pkg/nfd-master/nfd-master.go

```diff
@@ -1142,6 +1133,13 @@ func (m *nfdMaster) configure(filepath string, overrides string) error {
 	}
 	if m.args.Overrides.ResyncPeriod != nil {
 		c.ResyncPeriod = *m.args.Overrides.ResyncPeriod
+		// restart NFD API controller
+		if m.args.CrdController {
+			err := m.startNfdControllerApi()
+			if err != nil {
+				return nil
+			}
+		}
```

Comment:
I'd suggest to move the controller restart out of the `configure()` method, to the mainloop where the config watch event is processed